### PR TITLE
docs: document how to set OA-specific parameter options

### PR DESCRIPTION
**Pull request by lddubeau**
As requested in https://github.com/fastify/fastify-swagger/issues/178#issuecomment-582627553

This documents how to set Open API-specific parameter options in a schema used for parsing query strings.

#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message and code follows [Code of conduct](https://github.com/fastify/fastify/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -214,6 +214,50 @@ fastify.register(require('fastify-swagger'), {
...
+      properties: {
+        fields: {
+          type: 'array',
+          items: {
+            type: 'string'
+          },
+          minItems: 1,
+          //
+          // Note that this is an Open API version 2 configuration option.  The
+          // options changed in version 3.
```
Comment (mcollina) [inline README.md]: Good work! Which is the option for v3?

Comment (lddubeau) [reply README.md]: I decided to mention it without going into the gory details so that *if* someone tries it in v3 the note gives them fair warning. I decided against going into more details mainly because:

1. I think the plugin currently does not let us use v3. Mind you, the way I came to this conclusion was that I shoved the v3 options meant to do the same thing as I was doing in v2 in my configuration and it did nothing good. Then I eyeballed the code and [saw](https://github.com/fastify/fastify-swagger/blob/70461eeaa5ea7290b1549e98b7c03f3ac1511a15/dynamic.js#L62):

   ```
    swaggerObject.swagger = '2.0'
   ```

   I scanned the code and did not see how this specific field might be overwritten somehow. Hmm... actually now that I review this, I'm thinking that having Swagger run in v3 mode would require more than changing the value of this field. As I recall, `swagger` is not a valid field in v3 at all. When running in v3 mode, it should set another field, the `openapi` field I believe.

   Also, I did not find mention in the plugin documentation as to how to select an Open API version. Switching versions is a big deal because v3 is not backward compatible with v2. So the lack of documentation was (for me) another indication that probably switching versions is not possible right now.
   
2. Going into the details of how it works in v3 would require more explanation than I think is warranted for the issue at hand here. For one thing, in v3 the configuration which is equivalent to v2's `collectionFormat: "multi"` *is the default handling in v3*. (The default in v2 was `collectionFormat: "csv"`.) So in fact, a lack of further specification in v3 gives a behavior which is compatible with the default query parser. Stated explicitly in v3's options, v3's default is becomes: `style: "form", explode: true`. Ok, how do these two interact?

   Basically, I'm trying to avoid turning the README into a tutorial on how to use Open API.

Comment (mcollina) [reply README.md]: Great writeup, can you mention that this module supports v2 and not v3 in the comment?

Comment (lddubeau) [reply README.md]: Done.
